packages/kit/src/runtime/server/page/render.js

```diff
@@ -394,7 +412,12 @@ export async function render_response({
 				hydrate.push(`status: ${status}`);
 			}
 
-			if (options.embedded) {
+			if (manifest._.client.routes) {
+				if (event.route.id) {
+					const route = create_stringified_csr_server_route(event.route.id, event.url, manifest);
```

Comment:
If we do the lookup here, we don't need to do it inside `create_stringified_csr_server_route`, which allows us to simplify the types therein

```suggestion
					const route = manifest._.client.routes.find((r) => r.id === event.route.id);
					
					const stringified = create_stringified_csr_server_route(event.route.id, event.url, manifest);
```

Reply:
Similar to my comment above - I don't want to do that because then the server runtime logic about this stuff bleeds into other files, now it's all neatly contained within one file.

Reply:
It's further complicated by the fact that the `string | Route` is needed in two places inside `render.js`, so we would duplicated that "find route" logic.
We can create another function that just does `id -> Route` and use that in both places if you prefer that, but I feel like at that point it's easier to keep that stuff inside that function.

Reply:
I've often regretted functions with this sort of signature. It's never worth it, it's always better to just be explicit about what you're passing in. We can just find the route once, in the `if (page_config.csr)` block, and use it in both places, and it makes the code simpler and clearer. Will push a change up